code/game/machinery/cryopod.dm

```diff
@@ -0,0 +1,442 @@
+							to_chat(H.current, "<span class='danger'>Nar'Sie</span> murmurs, <span class='cultlarge'>[occupant] is beyond your reach. Sacrifice [GLOB.sac_mind.current] instead...</span></span>")
+
+	//Update any existing objectives involving this mob.
+	for(var/datum/objective/O in GLOB.objectives)
+		// We don't want revs to get objectives that aren't for heads of staff. Letting
+		// them win or lose based on cryo is silly so we remove the objective.
+		if(istype(O,/datum/objective/mutiny) && O.target == mob_occupant.mind)
+			qdel(O)
+		else if(O.target && istype(O.target, /datum/mind))
+			if(O.target == mob_occupant.mind)
+				if(O.owner && O.owner.current)
+					to_chat(O.owner.current, "<BR><span class='userdanger'>You get the feeling your target is no longer within reach. Time for Plan [pick("A","B","C","D","X","Y","Z")]. Objectives updated!</span>")
+				O.target = null
+				spawn(10) //This should ideally fire after the occupant is deleted.
+					if(!O)
+						return
+					O.find_target()
+					O.update_explanation_text()
+					if(!(O.target))
+						O.owner.objectives -= O
+						qdel(O)
+
+	if(mob_occupant.mind && mob_occupant.mind.assigned_role)
+		//Handle job slot/tater cleanup.
+		var/job = mob_occupant.mind.assigned_role
+		SSjob.FreeRole(job)
+		if(mob_occupant.mind.objectives.len)
+			mob_occupant.mind.objectives.Cut()
+			mob_occupant.mind.special_role = null
+
+	// Delete them from datacore.
+
+	var/announce_rank = null
+	for(var/datum/data/record/R in GLOB.data_core.medical)
+		if((R.fields["name"] == mob_occupant.real_name))
+			qdel(R)
+	for(var/datum/data/record/T in GLOB.data_core.security)
+		if((T.fields["name"] == mob_occupant.real_name))
+			qdel(T)
+	for(var/datum/data/record/G in GLOB.data_core.general)
+		if((G.fields["name"] == mob_occupant.real_name))
+			announce_rank = G.fields["rank"]
+			qdel(G)
+
+	for(var/obj/machinery/computer/cloning/cloner in world)
+		for(var/datum/data/record/R in cloner.records)
+			if(R.fields["name"] == mob_occupant.real_name)
+				cloner.records.Remove(R)
+
+	//Make an announcement and log the person entering storage.
+	if(control_computer)
+		control_computer.frozen_crew += "[mob_occupant.real_name]"
+
+	if(GLOB.announcement_systems.len)
+		var/obj/machinery/announcement_system/announcer = pick(GLOB.announcement_systems)
+		announcer.announce("CRYOSTORAGE", mob_occupant.real_name, announce_rank, list())
+		visible_message("<span class='notice'>\The [src] hums and hisses as it moves [mob_occupant.real_name] into storage.</span>")
+
+
+	for(var/obj/item/W in mob_occupant.GetAllContents())
+		if(W.loc.loc && (( W.loc.loc == loc ) || (W.loc.loc == control_computer)))
+			continue//means we already moved whatever this thing was in
+			//I'm a professional, okay
+		for(var/T in preserve_items)
+			if(istype(W, T))
+				if(control_computer && control_computer.allow_items)
+					control_computer.frozen_items += W
+					mob_occupant.transferItemToLoc(W, control_computer, TRUE)
```

Comment:
Now uses the proper unequip procs. Will hopefully make it not break layers, though I couldn't really replicate it myself. :v